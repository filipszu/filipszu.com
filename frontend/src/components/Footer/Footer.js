import classes from "./Footer.module.css";
import { BrowserView, MobileView } from 'react-device-detect';

const footer = (props) => {
    return (
        <footer className={classes.Footer}>
            <BrowserView>
                |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://twitter.com/filipszu">@filipszu</a>&nbsp;
                |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://github.com/filipszu">GitHub</a>&nbsp;
                |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://stackoverflow.com/users/9680496/filip-szulczewski">StackOverflow</a>&nbsp;
                |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://www.linkedin.com/in/filipszu">LinkedIn</a>&nbsp;|
            </BrowserView>
            <MobileView>
                |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://twitter.com/filipszu">@filipszu</a>&nbsp;
                |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://github.com/filipszu">GitHub</a>&nbsp;|
                <br />
                |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://stackoverflow.com/users/9680496/filip-szulczewski">StackOverflow</a>&nbsp;
                |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://www.linkedin.com/in/filipszu">LinkedIn</a>&nbsp;|
            </MobileView>
        </footer>
    );
};

export default footer;