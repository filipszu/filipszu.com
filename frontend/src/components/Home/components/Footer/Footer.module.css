.Footer{
	text-align: center;
	position: absolute;
	bottom: 0px;
	font-size: 32px;
	margin-bottom: 10px;
	width: 90%;	
}

.Footer a{
	text-decoration: none;
	border-bottom: 1px solid #A2EF00;
	color: #A2EF00; 	
}

.Footer a:hover{
	text-decoration: none;
	border-bottom: 1px solid #CB0077;
	color: #CB0077; 	
}

@media only screen and (max-width : 320px) {
	.Footer{
		font-size: 24px;
		line-height: 1;
	}
}

@media only screen and (max-width : 800px) {
	.Footer{
		font-size: 24px;
		line-height: 1;
	}
}