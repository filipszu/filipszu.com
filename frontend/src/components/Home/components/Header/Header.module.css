.Logo {
  font-size: 30px;
  margin: 0px auto;
}

.Logo h1,
.Logo h4 {
  margin: 0px;
  padding: 0px;
}

.Logo h4 {
  margin-top: -10px;
  padding-left: 49px;
  color: #a2ef00;
}

@media only screen and (max-width: 1366px) {
  .Logo h4 {
    font-size: 24px;
  }
}
