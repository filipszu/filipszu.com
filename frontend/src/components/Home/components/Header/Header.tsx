import classes from "./Header.module.css";
export interface HeaderProps {
  title?: string;
  subtitle?: string;
}
const Header = (props: HeaderProps) => {
  const title = props.title || "FiliPSZU";
  const subtitle = props.subtitle || "Software Engineer";
  return (
    <div className={classes.Logo}>
      <h1>{title}</h1>
      <h4>{subtitle}</h4>
    </div>
  );
};
export default Header;
