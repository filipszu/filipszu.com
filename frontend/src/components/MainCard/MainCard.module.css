.MainCard {
    margin: 10% auto;
	min-width: 400px;
	min-height: 350px;
	width: 40%;
	height: 25%;
	border: 5px dashed #CB0077;
	z-index: 10;
	position: relative;
	background-color: rgba(0,0,0,0.85);
	top: 30%;
	padding: 10px 15px 0px 15px;
}

.TextWhite{
	color: #FFFFFF;
}
.TextGreen{
	color: #A2EF00;
}
.TextRed{
	color: #AA0000;
}
.TextPurple{
	color: #CB0077;
}

.Content {
	margin-bottom: 80px;
	margin-top: 40px;
}

.AboutText {
	color: #999999;
	text-align: left;
	font-size: 24px;
	display: inline;
}

.Logo {
    font-size: 30px;
	margin: 0px auto;  
}

.Logo h1, 
.Logo h4{
	margin: 0px;
	padding:0px;
}

.Logo h4{
	margin-top: -10px;
	padding-left: 49px;
	color: #A2EF00;
}

@media only screen and (max-width : 800px) {
	.MainCard {
		min-width: 280px;
		min-height: 480px;
		height: 100%;
		margin-top: 10px;
		padding: 15px 15px 10px 15px;
	}
}

@media only screen and (max-width : 320px) {
	.MainCard {
		min-width: 270px;
		min-height: 480px;
		height: 100%;
		margin-top: 10px;
		padding: 5px 5px 5px 5px;			
	}
}