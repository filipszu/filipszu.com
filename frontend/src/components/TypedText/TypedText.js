import classes from './TypedText.module.css';
import React, { useEffect, useState } from 'react';

const TypedText = (props) => {
    const requestRef = React.useRef();
    const previousTimeRef = React.useRef();
    const lastAnimationTickRef = React.useRef();
    const isDoneAnimating = React.useRef();
    const [isAnimating, setIsAnimating] = useState(false);
    const [typedText, setTypedText] = useState('');

    const type = () => {
        setTypedText(prevTypedText => { 
            const newLength = prevTypedText.length+1;
            const newValue = props.children.substr(0, newLength);
            if(newLength === props.children.length){
                isDoneAnimating.current = true;
            }
            return newValue;
        });
    };

    const animate = time => {
        if (previousTimeRef.current !== undefined) {
            
            setIsAnimating(prevAnimating => {
                if(!prevAnimating && time >= props.delay){
                    lastAnimationTickRef.current = time;
                    type();
                    return true;
                }
    
                if(prevAnimating && time - lastAnimationTickRef.current >= props.interval){
                    lastAnimationTickRef.current = time;
                    type();
                }

                return prevAnimating;
            });
          }

          previousTimeRef.current = time;
          if(!isDoneAnimating.current){
            requestRef.current = requestAnimationFrame(animate);
          }
    };

    useEffect(() => {
        requestRef.current = requestAnimationFrame(animate);
        return () => cancelAnimationFrame(requestRef.current);
    }, []);
    
    const cursorClasses = [classes.Pulse, classes.TextCursor].join(' ');
    return (
        <div className={props.className}>
            <p>
                <span dangerouslySetInnerHTML={{__html: typedText}}></span><span className={cursorClasses}>â–®</span>
            </p>
        </div>
    );
};

export default TypedText;