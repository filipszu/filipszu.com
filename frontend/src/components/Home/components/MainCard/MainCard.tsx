import classes from "./MainCard.module.css";

import Footer from "../Footer/Footer";
import TypedText from "../TypedText/TypedText";
import Header from "../Header/Header";

export interface MainCardProps {
  aboutText?: string;
}

export const TestID = "szu-maincard";

const MainCard = (props: MainCardProps) => {
  const aboutText = props.aboutText || "Welcome to the main card!";
  return (
    <div className={classes.MainCard} data-testid={TestID}>
      <Header data-testid="main-card-header" />
      <section className={classes.Content}>
        <TypedText
          data-testid="main-card-typedtext"
          className={classes.AboutText}
          delay={500}
          interval={10}
          text={aboutText}
        />
      </section>
      <Footer data-testid="main-card-footer" />
    </div>
  );
};

export default MainCard;
