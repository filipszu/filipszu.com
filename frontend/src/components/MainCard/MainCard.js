import classes from './MainCard.module.css';
import { useState } from 'react';

import Footer from '../Footer/Footer';
import TypedText from '../TypedText/TypedText';

const MainCard = (props) => {
    const [aboutText, setAboutText] = useState(`Hi,<br />
        my name is <span class="${classes.TextWhite}">Filip Szulczewski</span>.<br />
        I’m an <span class="${classes.TextGreen}">autodidact programmer</span>, with <span class="${classes.TextWhite}">more than a decade of experience</span>.<br /><br />  
        I’ve <span class="${classes.TextGreen}">architected</span>, <span class="${classes.TextGreen}">built</span>, and <span class="${classes.TextGreen}">delivered</span> an array of <span class="${classes.TextPurple}">video applications</span> using <span class="${classes.TextPurple}">web technologies</span>. 
        I have a <span class="${classes.TextWhite}">track record</span> of <span class="${classes.TextWhite}">succesfully augmenting teams</span> in <span class="${classes.TextWhite}">Professional Services</span> capacity.  
        I've worked both on <span class="${classes.TextPurple}">frontend</span> and <span class="${classes.TextPurple}">backend</span> projects.  
        I’m a <span class="${classes.TextPurple}">FOSS</span> enthusiast with a <span class="${classes.TextPurple}">DIY</span> spirit.`);

    return (
        <div className={classes.MainCard}>
            <div className={classes.Logo}>
                <h1>FiliPSZU</h1>
                <h4>Software Engineer</h4>
			</div>
            <section className={classes.Content}>
                <TypedText className={classes.AboutText} delay={500} interval={10}>{aboutText}</TypedText>
            </section>
            <Footer />
        </div>
    );
};

export default MainCard;