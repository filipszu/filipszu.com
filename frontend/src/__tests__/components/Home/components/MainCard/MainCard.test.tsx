import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import MainCard, {
  TestID,
} from "../../../../../components/Home/components/MainCard/MainCard";
import { TestID as HeaderTestID } from "../../../../../components/Home/components/Header/Header";
import { TestID as FooterTestID } from "../../../../../components/Home/components/Footer/Footer";
import { TestID as TypedTextID } from "../../../../../components/Home/components/TypedText/TypedText";

describe(`MainCard component responsibble for showing the Title, TypedText and Footer`, () => {
  test(`It will render correctly and match the snapshot.`, () => {
    const tree = renderer.create(<MainCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test(`[Given] it will render [Then] it will use the appropriate test-id: ${TestID}`, () => {
    render(<MainCard />);
    const el = screen.getByTestId(TestID);
    expect(el).toBeInTheDocument();
  });
  test(`[Given] it will render [Then] it will render the Title Component`, () => {
    render(<MainCard />);
    const el = screen.getByTestId(HeaderTestID);
    expect(el).toBeInTheDocument();
  });
  test(`[Given] it will render [Then] it will render the TypedText Component`, () => {
    render(<MainCard />);
    const el = screen.getByTestId(TypedTextID);
    expect(el).toBeInTheDocument();
  });
  test(`[Given] it will render [Then] it will render the Footer Component`, () => {
    render(<MainCard />);
    const el = screen.getByTestId(FooterTestID);
    expect(el).toBeInTheDocument();
  });
});
