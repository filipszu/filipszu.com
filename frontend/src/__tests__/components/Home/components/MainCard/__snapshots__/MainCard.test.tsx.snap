// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainCard component responsibble for showing the Title, TypedText and Footer It will render correctly and match the snapshot. 1`] = `
<div
  className="MainCard"
  data-testid="szu-maincard"
>
  <div
    className="Logo"
    data-testid="szu-header"
  >
    <h1>
      FiliPSZU
    </h1>
    <h4>
      Software Engineer
    </h4>
  </div>
  <section
    className="Content"
  >
    <div
      className="AboutText"
      data-testid="szu-typedtext"
    >
      <p>
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
        <span
          className="Pulse TextCursor"
        >
          ▮
        </span>
      </p>
    </div>
  </section>
  <footer
    className="Footer"
  >
    <div
      data-testid="szu-footer"
    >
      | 
      <a
        href="https://twitter.com/filipszu"
        rel="nofollow noopener noreferrer"
        target="_blank"
      >
        @filipszu
      </a>
        | 
      <a
        href="https://github.com/filipszu"
        rel="nofollow noopener noreferrer"
        target="_blank"
      >
        GitHub
      </a>
        | 
      <a
        href="https://stackoverflow.com/users/9680496/filip-szulczewski"
        rel="nofollow noopener noreferrer"
        target="_blank"
      >
        StackOverflow
      </a>
        | 
      <a
        href="https://www.linkedin.com/in/filipszu"
        rel="nofollow noopener noreferrer"
        target="_blank"
      >
        LinkedIn
      </a>
       |
    </div>
  </footer>
</div>
`;
